src/bridge/AdminRoomHandler.ts

```diff
@@ -39,6 +39,15 @@ enum CommandPermission {
 // This is just a length to avoid silly long usernames
 const SANE_USERNAME_LENGTH = 64;
 
+// Technically, anything but \0 is allowed as username (aka. authcid and authzid):
+// https://www.rfc-editor.org/rfc/rfc4616#section-2
+//
+// However, IRC services are very unlikely to allow the username to contain CR or LF
+// (because they would not fit in the wire format) or spaces (because usernames are
+// usually followed by passwords in "PRIVMSG NickServ :REGISTER" commands and
+// IRCv3 draft/account-registration.
+const SASL_USERNAME_INVALID_CHARS_PATTERN = /[ \0\r\n]+/;
```

Comment:
Wondering if we at least limit to printable characters. I can't see the value in allowing most of these.

```suggestion
const SASL_USERNAME_INVALID_CHARS_PATTERN = /[\0-\38\7F]+/;
```

Reply:
I think you mean `[\0-\40\x7F]` (38 isn't valid octal, the octal for space is 40 and 7F is the hex code for DEL, not the octal code).

Reply:
yeah, bad copy paste. That one ^